pkg/controller/importconfig/cluster_info.go

```diff
@@ -144,3 +109,183 @@ func validateToken(token string, creation, expiration []byte) bool {
 	lifetime := time.Until(expirationTime)
 	return lifetime > refreshThreshold
 }
+
+func buildBootstrapKubeconfigData(ctx context.Context, clientHolder *helpers.ClientHolder,
+	managedCluster *clusterv1.ManagedCluster,
+	klusterletConfig *klusterletconfigv1alpha1.KlusterletConfig) ([]byte, []byte, []byte, error) {
+	var bootstrapKubeconfigData, tokenData, tokenCreation, tokenExpiration []byte
```

Comment:
I thought the internal endpoint kubeconfig has a fixed format, so we can return it directly and keep the other logic unchanged. like:
```
if isSelfManaged(managedCluster) {
  return internal-kubeconfig-bootstrap-config-data,....
}

...
```

Reply:
The internal endpoint will be used only when:
- The managed cluster is the local-clsuter;
- And there is no custom config of kube-apiserver url and CA;

So the user is still able to change the kube-apiserver url or CA of a self managed cluster by using the klusterletconfig API.

Reply:
This PR also refactors the logic to create the bootstrap kubeconfig. It becomes simpler and uses less API calls to the apiserver now.